### DISCUSSION TITLE
Nuclei having issues with multiple sessions

### DISCUSSION THREAD

**Discussion by inthenightsky** [projectdiscovery/nuclei#1544] [Q&A] [answered]
### Nuclei version:

2.5.8

### Current Behavior:
When running Nuclei across multiple processes (i.e. via screen instances on Linux) and launching Nuclei with the -rdb <folder name>, it will refuse to launch until the other session has ended, stating:

```
[FTL] Could not create runner: could not create issue reporting client: resource temporarily unavailable
```

### Expected Behavior:
Nuclei should launch as normal, reporting into the report database.

### Steps To Reproduce:
Open a new screen session on Linux, and kickoff a Nuclei scan using the -rdb flag.
Detach from that screen session (CTRL + A + D), the first scan will still be running.
Now attempt to run another Nuclei scan using the -rdb flag.
Observe the error occur.

### Anything else:
This is likely because the process is locking the directory whilst the scan is in progress, so the other process running Nuclei is unable to access it. A solution might be to launch a separate process for interacting with the report database, have that process lock the directory, and have Nuclei route all requests to the report database through that process instead.

**Comment by ehsandeep** [comment, marked as answer]
@ImNightmaree could you confirm if you able to reproduce this issuewith unique `-rdb` directory for each scan as expected?

**Reply by forgedhallpass** [reply]
@ImNightmaree since we haven't heard back from you, I am marking the response above as the answer to this question.

**Comment by wifiuk** [comment]
i get this issue even using the -rdb flag when running as root, works fine when running as non root user in kali.
![image](https://github.com/user-attachments/assets/7c2fe0df-7fc5-4cac-9893-972c4d464d22)
